lib/libsm/sm3.c

```diff
@@ -0,0 +1,290 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright Copyright (c) 2019 Huawei Technologies Co., Ltd
+ */
+/*
+ * SM3 Hash algorithm
+ * thanks to Xyssl
+ * author:goldboar
+ * email:goldboar@163.com
+ * 2011-10-26
+ */
+
+#include <sm3.h>
+#include <string.h>
+#include <string_ext.h>
+
+#define GET_UINT32_BE(n, b, i)			\
+{						\
+	(n) = ((uint32_t)(b)[(i)] << 24)	\
+	    | ((uint32_t)(b)[(i) + 1] << 16)	\
+	    | ((uint32_t)(b)[(i) + 2] <<  8)	\
+	    | ((uint32_t)(b)[(i) + 3]);		\
+}
+
+#define PUT_UINT32_BE(n, b, i)			\
+{						\
+	(b)[(i)] = (uint8_t)((n) >> 24);	\
+	(b)[(i) + 1] = (uint8_t)((n) >> 16);	\
+	(b)[(i) + 2] = (uint8_t)((n) >>  8);	\
+	(b)[(i) + 3] = (uint8_t)((n));		\
+}
+
+void sm3_init(struct sm3_context *ctx)
+{
+	ctx->total[0] = 0;
+	ctx->total[1] = 0;
+
+	ctx->state[0] = 0x7380166F;
+	ctx->state[1] = 0x4914B2B9;
+	ctx->state[2] = 0x172442D7;
+	ctx->state[3] = 0xDA8A0600;
+	ctx->state[4] = 0xA96F30BC;
+	ctx->state[5] = 0x163138AA;
+	ctx->state[6] = 0xE38DEE4D;
+	ctx->state[7] = 0xB0FB0E4E;
+}
+
+static void sm3_process(struct sm3_context *ctx, const uint8_t data[64])
+{
+	uint32_t SS1, SS2, TT1, TT2, W[68], W1[64];
+	uint32_t A, B, C, D, E, F, G, H;
+	uint32_t T[64];
+	uint32_t Temp1, Temp2, Temp3, Temp4, Temp5;
+	int j;
+
+	for (j = 0; j < 16; j++)
+		T[j] = 0x79CC4519;
+	for (j = 16; j < 64; j++)
+		T[j] = 0x7A879D8A;
+
+	GET_UINT32_BE(W[0], data,  0);
+	GET_UINT32_BE(W[1], data,  4);
+	GET_UINT32_BE(W[2], data,  8);
+	GET_UINT32_BE(W[3], data, 12);
+	GET_UINT32_BE(W[4], data, 16);
+	GET_UINT32_BE(W[5], data, 20);
+	GET_UINT32_BE(W[6], data, 24);
+	GET_UINT32_BE(W[7], data, 28);
+	GET_UINT32_BE(W[8], data, 32);
+	GET_UINT32_BE(W[9], data, 36);
+	GET_UINT32_BE(W[10], data, 40);
+	GET_UINT32_BE(W[11], data, 44);
+	GET_UINT32_BE(W[12], data, 48);
+	GET_UINT32_BE(W[13], data, 52);
+	GET_UINT32_BE(W[14], data, 56);
+	GET_UINT32_BE(W[15], data, 60);
+
+#define FF0(x, y, z) ((x) ^ (y) ^ (z))
+#define FF1(x, y, z) (((x) & (y)) | ((x) & (z)) | ((y) & (z)))
+
+#define GG0(x, y, z) ((x) ^ (y) ^ (z))
+#define GG1(x, y, z) (((x) & (y)) | ((~(x)) & (z)))
+
+#define SHL(x, n)  (((x) & 0xFFFFFFFF) << (n))
+#define ROTL(x, n) (SHL((x), n) | ((x) >> (32 - n)))
```

Comment:
```patch
-#define ROTL(x, n) (SHL((x), n) | ((x) >> (32 - n)))
+#define ROTL(x, n) (SHL((x), (n)) | ((x) >> (32 - (n))))
```

Would be easier to read with a tabulation before the macro values.

Reply:
Agreed